Fix FancyArrow rendering for zero-length arrows

nrnavaneet opened the pull request:
Fixes [#30242](https://github.com/matplotlib/matplotlib/issues/30242): Cannot create empty FancyArrow (expired NumPy deprecation).

Summary of changes:
	•	Addresses a bug where FancyArrow(length=0) caused issues due to expired NumPy behavior.
	•	Adds an explicit check for zero length to safely initialize self.verts as an empty array.
	•	Adds a test case test_empty_fancyarrow in test_patches.py to confirm correct behavior for zero-length arrows.

This change ensures that FancyArrow gracefully handles length=0 without triggering errors or deprecated behavior in newer NumPy versions.

QuLogic commented on lib/matplotlib/patches.py:
```diff
@@ -1538,7 +1538,7 @@ def _make_verts(self):
         else:
             length = distance + head_length
-        if not length:
+        if (isinstance(length, np.ndarray) and length.size == 0) or (not length):
```
You can use `np.size` here.

QuLogic commented on .python-version:
```diff
@@ -0,0 +1 @@
+3.10.13
```
Don't add this file.

nrnavaneet commented:
Thank you for the review, @QuLogic! 🙌
I’ve updated the condition to use np.size(length) as suggested, and also removed the .python-version file from the commit. Everything should be good now. Please let me know if anything else needs changes. 😊
